src/plugins/data/common/search/expressions/query_filter.ts

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { isNil, omit, omitBy } from 'lodash';
+import { i18n } from '@kbn/i18n';
+import { ExpressionFunctionDefinition, ExpressionValueBoxed } from 'src/plugins/expressions/common';
+import { Query } from '../../query';
+import { KibanaQueryOutput } from './kibana_context_type';
+
+export interface QueryFilter {
+  input: Query;
+  label?: string;
+}
+
+export type QueryFilterOutput = ExpressionValueBoxed<'kibana_query_filter', QueryFilter>;
+
+interface QueryFilterArguments {
+  input: KibanaQueryOutput;
+  label?: string;
+}
+
+export type ExpressionFunctionQueryFilter = ExpressionFunctionDefinition<
+  'queryFilter',
+  null,
+  QueryFilterArguments,
+  QueryFilterOutput
+>;
+
+export const queryFilterFunction: ExpressionFunctionQueryFilter = {
+  name: 'queryFilter',
```

Comment:
I'm confused by what this datatype represents. It seems to represent a label + query, but it's called `queryFilter`, and it takes a `query` as input...?

Reply:
I called that based on the UI. In the UI, we have a query filter that consists of the query itself and an optional label. I could not come up with a better name. Maybe there is a better name we can use here?

Reply:
its a function that generate a `query` filter.

Reply:
But `query` and `filter` are synonyms. I suggest that you name this `labeledFilter` or `labeledQuery`.